### PR TITLE
Service Registry access rules

### PR DESCRIPTION
**Pull request by wtrocki**
Still WIP but functional. Creating for feedback/binary across teams

## Plan 

 

Then we need specific commands for  registry to manage rbac

```bash
## No difference for us if we create or update roles
rhoas service-registry role add --user=test --role=WRITE
rhoas service-registry role revoke --user=test
rhoas service-registry role list 
```

We might look for the ability to create a new service account with roles 
For simplicity, we can even allow creating SA and role assignment in a single command

```bash
rhoas service-registry role add --new --role=WRITE
```

For simplicity, we can give the user the ability to set a role for himself without specifying the name

```bash
rhoas service-registry role add --me --role=WRITE
```

> NOTE The creation of the service account will give us only read access to topics, consumer groups etc.
Additional actions will be needed for CLI, Operator to supply extra roles for the service account.

### CODE REVIEW

File: pkg/cmd/registry/artifact/crud/create/create.go [hunk 1]
```diff
@@ -73,6 +73,12 @@ func NewCreateCommand(f *factory.Factory) *cobra.Command {
 				opts.file = args[0]
 			}
 
+			if opts.artifactType != "" {
+				if _, err := registryinstanceclient.NewArtifactTypeFromValue(opts.artifactType); err != nil {
+					return errors.New(opts.localizer.MustLocalize("artifact.cmd.create.error.invalidArtifactType", localize.NewEntry("AllowedTypes", util.GetAllowedArtifactTypeEnumValuesAsString())))
```
Comment (craicoverflow) [inline pkg/cmd/registry/artifact/crud/create/create.go:78]:
```suggestion
					return errors.New(opts.localizer.MustLocalize("artifact.cmd.create.error.invalidArtifactType", localize.NewEntry("AllowedTypes", util.GetAllowedArtifactTypeEnumValuesAsString())))
```
```suggestion
					return opts.localizer.MustLocalizeError("artifact.cmd.create.error.invalidArtifactType", localize.NewEntry("AllowedTypes", util.GetAllowedArtifactTypeEnumValuesAsString()))
```

Comment (wtrocki) [reply pkg/cmd/registry/artifact/crud/create/create.go:78]: Good catch. Will fix that in couple places.

**Comment by wtrocki** [comment]
Blocked because API seems to be not there yet - also we are missing ways to fetch principals and validate them.
Leaving as blocked for now

**Comment by wtrocki** [comment]
Created new issue to cover blocking parts. This way we can merge this changes and resolve issue later phase:
https://github.com/redhat-developer/app-services-cli/issues/1095

**Comment by wtrocki** [comment]
I would leave org list command for the moment. The reason why this was blocked were lack of ability to list user principals. 
Api will improve over the time and we can fix that properly.

File: pkg/cmd/registry/artifact/role/list/list.go [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+		IO:         f.IOStreams,
+		localizer:  f.Localizer,
+		Context:    f.Context,
+	}
+
+	cmd := &cobra.Command{
+		Use:     "list",
+		Short:   f.Localizer.MustLocalize("registry.role.cmd.list.shortDescription"),
+		Long:    f.Localizer.MustLocalize("registry.role.cmd.list.longDescription"),
+		Example: f.Localizer.MustLocalize("registry.role.cmd.list.example"),
```
Comment (rkpattnaik780) [inline pkg/cmd/registry/artifact/role/list/list.go:59]:
```suggestion
		Example: f.Localizer.MustLocalize("registry.role.cmd.list.example"),
		Args:    cobra.NoArgs,
```

File: pkg/localize/locales/en/cmd/role.toml [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+When revoking role of the owner of the instance user will still have ADMIN rights.
+'''
+ 
+[registry.role.cmd.revoke.example] 
+one = '''
+## Revoke role for user
+rhoas service-registry role revoke --username=janedough
+''' 
+
+[registry.role.cmd.revoke.success]
```
Comment (rkpattnaik780) [inline pkg/localize/locales/en/cmd/role.toml:101]: Nit: should we elaborate this message?

**Review by rkpattnaik780 — CHANGES_REQUESTED** [review]
Role list command doesnt work for json and yaml output formats.
Screenshot from 2021-09-22 14-26-11

File: pkg/localize/locales/en/cmd/role.toml [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+'''
+ 
+[registry.role.cmd.revoke.example] 
+one = '''
+## Revoke role for user
+rhoas service-registry role revoke --username=janedough
+''' 
+
+[registry.role.cmd.revoke.success] 
+one = 'Success'
```
Comment (wtrocki) [inline pkg/localize/locales/en/cmd/role.toml:102]:
```suggestion
one = 'Successfully revoked access for specified principal' 
```

**Comment by wtrocki** [comment]
> Role list command doesnt work for json and yaml output formats.

This must have been regression in other PR. I will also check how other lists commands work now.

**Comment by wtrocki** [comment]
Actually other commands work. it is that I have old version of the code that weren't changed

**Comment by wtrocki** [comment]
Merging to create srs release for tommorow

**Pull request merged**